File: Data Structures & Algorithms/binary-search/submission-4.java
class Solution {
    public int search(int[] nums, int target) {
        int n = nums.length;
        int L = 0;
        int R = n - 1;

        while(L <= R){
            int mid = (R + L) / 2;

            if(nums[mid] == target){
                return mid;
            } else if(nums[mid] > target){
                R--;
            } else{
                L++;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-4.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        L = 0
        total = 0
        minL = float("inf")

        for R in range(n):
            total += nums[R]

            while total >= target:
                minL = min(minL, R - L + 1)
                total -= nums[L]
                L += 1
        return 0 if minL == float('inf') else minL